pkg/controller/account/account_controller.go

```diff
@@ -323,12 +319,105 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
+			r.getManagedTags(reqLogger),
+			r.getCustomTags(reqLogger, currentAcctInstance),
+		).GetIAMTags()
+
+		// In this block we are creating the BYOCAdminAccessRole-XYZ for both CCS and non-CCS accounts.
+		// The dependancy on the roleID validation within the assumeRole, and the different aws clients
+		// required between CCS and non-CCS, is what has caused these steps to be done independantly.
+		if currentAcctInstance.Spec.BYOC {
+			// The CCS uses the CCS client for creating the BYOCAdminAccessRole and then utilizes the RoleID
+			// generated from that in the assumeRole func for role validation
+
+			// Get the AccountClaim in Order to retrieve the CCSClient
+			accountClaim, acctClaimErr := r.getAccountClaim(currentAcctInstance)
+			if acctClaimErr != nil {
+				if accountClaim != nil {
+					utils.SetAccountClaimStatus(
+						accountClaim,
+						"Failed to get AccountClaim for Account",
+						"FailedRetrievingAccountClaim",
+						awsv1alpha1.ClientError,
+						awsv1alpha1.ClaimStatusError,
+					)
+					err := r.Client.Status().Update(context.TODO(), accountClaim)
+					if err != nil {
+						reqLogger.Error(err, "failed to update accountclaim status")
+					}
+				} else {
+					reqLogger.Error(acctClaimErr, "accountclaim is nil")
+				}
+				return reconcile.Result{}, acctClaimErr
+			}
+			ccsClient, err := r.getCCSClient(currentAcctInstance, accountClaim)
+			if err != nil {
+				reqLogger.Error(err, "An error was encountered retrieving CCS Client")
+				return reconcile.Result{}, err
+			}
+
+			roleID, err = createBYOCAdminAccessRole(
+				reqLogger,
+				awsSetupClient,
+				ccsClient,
+				adminAccessArn,
+				currentAccInstanceID,
+				tags,
+				SREAccessARN,
+			)
+
+			if err != nil {
+				reqLogger.Error(err, "createBYOCAdminAccessRole err", roleID)
+				return reconcile.Result{}, err
+			}
+
+			awsAssumedRoleClient, creds, err = r.assumeRole(reqLogger, currentAcctInstance, awsSetupClient, roleToAssume, roleID)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+
+		} else {
+			// Unlike the CCS block, the non-CCS block does not have a dependency on the RoleID to assumeRole. The
+			// awsAssumedRoleClient is what is needed to create the BYOCAdminAccessRole in the non-CCS account.
+			awsAssumedRoleClient, creds, err = r.assumeRole(reqLogger, currentAcctInstance, awsSetupClient, roleToAssume, "")
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+
+			roleID, err := createBYOCAdminAccessRole(
+				reqLogger,
+				awsSetupClient,
+				awsAssumedRoleClient,
+				adminAccessArn,
+				currentAccInstanceID,
+				tags,
+				SREAccessARN,
+			)
+
+			if err != nil {
+				reqLogger.Error(err, "createBYOCAdminAccessRole err", roleID)
+				return reconcile.Result{}, err
+			}
+		}
+
+		// Use the same ID applied to the account name for IAM usernames
+		iamUserUHC := fmt.Sprintf("%s-%s", iamUserNameUHC, currentAccInstanceID)
+		iamUserSRE := fmt.Sprintf("%s-%s", iamUserNameSRE, currentAccInstanceID)
```

Comment:
this user is already deleted at https://github.com/openshift/aws-account-operator/pull/607/files#diff-998c9da647c44633b8acec24f0b19b1446779dd895f2a2a8f6e4c6a56bc89b05L332 so I guess a rebase is needed